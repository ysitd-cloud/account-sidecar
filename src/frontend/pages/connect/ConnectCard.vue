<template>
    <v-card>
        <v-card-text class="pt-5">
            <v-icon class="headline">{{ icon }}</v-icon>
        </v-card-text>
        <v-card-title class="justify-center mt-3 pb-0">
            <h3 class="headline mb-0">{{ name }}</h3>
        </v-card-title>
        <v-card-actions class="justify-center">
            <v-btn flat color="primary" :href="url">Connect</v-btn>
        </v-card-actions>
    </v-card>
</template>

<style scoped lang="stylus">
    .card_title
        text-align center

    .card__text
        text-align: center
        .icon
            font-size 7rem !important
            color black
</style>

<script>
    export default {
      name: 'ConnectCard',
      props: {
        id: {
          type: String,
          required: true,
        },
        icon: {
          type: String,
          required: true,
        },
        name: {
          type: String,
          required: true,
        },
      },
      computed: {
        url() {
          return `/connect/${this.id}`;
        },
      },
    };
</script>
