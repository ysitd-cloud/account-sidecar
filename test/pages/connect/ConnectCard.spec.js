import { shallow } from 'vue-test-utils';
import { createRenderer } from 'vue-server-renderer';

import ConnectCard from '../../../src/frontend/pages/connect/ConnectCard.vue';

/* global describe, it, expect */

describe('ConnectCard.vue', () => {
  it('matches snapshot', () => {
    const renderer = createRenderer();
    const wrapper = shallow(ConnectCard, {
      propsData: {
        id: 'foo',
        icon: 'bar',
        name: 'Test',
      },
    });

    renderer.renderToString(wrapper.vm, (e, str) => {
      if (e) throw new Error(e);
      expect(str).toMatchSnapshot();
    });
  });
});
