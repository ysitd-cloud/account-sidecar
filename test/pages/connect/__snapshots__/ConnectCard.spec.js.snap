// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectCard.vue matches snapshot 1`] = `
<v-card>
    <v-card-text class="pt-5">
        <v-icon class="headline">bar</v-icon>
    </v-card-text>
    <v-card-title class="justify-center mt-3 pb-0">
        <h3 class="headline mb-0">Test</h3>
    </v-card-title>
    <v-card-actions class="justify-center">
        <v-btn flat="" color="primary" href="/connect/foo">Connect</v-btn>
    </v-card-actions>
</v-card>
`;
